### ISSUE TITLE
Incorrect Socks4 config

### ISSUE THREAD

**Issue by chromer030** [MatsuriDayo/nekoray#409]
**描述问题**

预期行为：
Nekobox is failing at connection to socks4 server and it's due to it's incorrect build-in config.
Firstly reported here : https://github.com/SagerNet/sing-box/issues/373
based on nekobox config, it's behaviouring Socks4 as Socks5 and this is why it's failing.
```
{"dns":{"rules":[{"domain":["geoip:de"],"domain_keyword":[],"domain_regex":[],"domain_suffix":[],"geosite":[],"server":"dns-direct"}],"servers":[{"address":"https://8.8.8.8/dns-query","address_resolver":"dns-underlying","detour":"proxy","strategy":"","tag":"dns-remote"},{"address":"local","address_resolver":"dns-underlying","detour":"direct","strategy":"","tag":"dns-direct"},{"address":"local","detour":"direct","tag":"dns-underlying"}]},"inbounds":[{"listen":"0.0.0.0","listen_port":1080,"sniff":true,"sniff_override_destination":false,"tag":"mixed-in","type":"mixed"}],"log":{"level":"info"},"outbounds":[{"domain_strategy":"","server":"127.0.0.1","server_port":1080,"tag":"proxy","type":"socks"},{"tag":"direct","type":"direct"},{"tag":"bypass","type":"direct"},{"tag":"block","type":"block"},{"tag":"dns-out","type":"dns"}],"route":{"final":"proxy","rules":[{"outbound":"dns-out","protocol":"dns"},{"geoip":[],"ip_cidr":["192.168.0.0/8"],"outbound":"bypass"},{"domain":["geoip:de"],"domain_keyword":[],"domain_regex":[],"domain_suffix":[],"geosite":[],"outbound":"bypass"}]}}
```

I tested this simple config in singbox with same server , it's working :
```
{
  "log": {
    "level": "info"
  },
  "dns": {
    "servers": [
      {
        "address": "tcp://1.0.0.1"
      }
    ]
  },
  "inbounds": [
    {
      "type": "mixed",
      "listen": "127.0.0.1",
      "listen_port": 1080
    }
  ],
  "outbounds": [
    {
		"type": "socks",
		"tag": "socks-out",
		"server": "IP",
		"server_port": PORT,
		"version": "4",
		"network": "tcp",
		"udp_over_tcp": false
    },
    {
      "type": "dns",
      "tag": "dns-out"
    }
  ],
  "route": {
    "rules": [
      {
        "protocol": "dns",
        "outbound": "dns-out"
      }
    ]
  }
}

```

实际行为：

**如何复现**

Create a new socks4 profile and test.

**日志**

如果有日志，请上传。请在文档内查看导出日志的详细步骤。

**Comment by chromer030** [comment]
UPDATE :
fix is just adding  `"version": "4",` after `"type":"socks"`

**Comment by chromer030** [comment]
Fixed in 2.18 , Thanks.
